lib/loader/ast/instruction.cpp

```diff
@@ -283,13 +283,7 @@ Expect<void> Loader::loadInstruction(AST::Instruction &Instr) {
 
   // Reference Instructions.
   case OpCode::Ref__null: {
-    Expect<RefType> Res;
-    if (Conf.hasProposal(Proposal::FunctionReferences)) {
-      // The func-ref proposal use the heap type to replace ref type here.
-      Res = loadHeapType(RefTypeCode::RefNull, ASTNodeAttr::Instruction);
-    } else {
-      Res = loadRefType(ASTNodeAttr::Instruction);
-    }
+    auto Res = loadRefType(ASTNodeAttr::Instruction);
```

Comment:
I will change to use `loadHeapType` in the next PR https://github.com/WasmEdge/WasmEdge/pull/2125. Because in the next PR I will switch to store `HeapType` instead of `RefType` in the `ref.null` instruction.

If we look at the binary code, the `loadHeapType` after supporting is a superset of `loadRefType` in the current stable spec. After we switch to use `loadHeapType`, we call add a check that if func-ref proposal is not enabled, the loaded heap type can only be `func` and `extern`.